apps/dashboard/lib/trpc/routers/key/updateRootKeyName.ts

```diff
@@ -0,0 +1,67 @@
+import { db, eq, schema } from "@/lib/db";
+import { env } from "@/lib/env";
+import { ingestAuditLogs } from "@/lib/tinybird";
+import { TRPCError } from "@trpc/server";
+import { z } from "zod";
+import { auth, t } from "../../trpc";
+
+export const updateRootKeyName = t.procedure
+  .use(auth)
+  .input(
+    z.object({
+      keyId: z.string(),
+      name: z.string().nullish(),
+    }),
+  )
+  .mutation(async ({ input, ctx }) => {
+    const key = await db.query.keys.findFirst({
+      where: (table, { eq, isNull, and }) =>
+        and(eq(table.id, input.keyId), isNull(table.deletedAt)),
+      with: {
+        workspace: true,
+      },
+    });
+
+    if (!key || key.ownerId !== ctx.tenant.id) {
```

Comment:
You need to check the workspace tenant is
Not owner id

Reply:
Get off the internet. I am reviewing the code

Reply:
@chronark the workspace tenant on the root key is the generic one, and will never match the `ctx.tenantId` -- it matches the tenant of the workspace associated to the root key's `for_workspaceId` 

Unless I grab the workspace that matches the key.for_workspaceId, and compare its tenantId to ctx.tenantId.